language/move-core/types/src/state.rs

```diff
@@ -12,14 +10,15 @@ pub enum VMState {
     OTHER,
 }
 
-thread_local! {
-    static STATE: RefCell<VMState> = RefCell::new(VMState::OTHER);
-}
+static mut STATE: VMState = VMState::OTHER;
 
 pub fn set_state(state: VMState) -> VMState {
-    STATE.with(|s| s.replace(state))
+    unsafe {
```

Comment:
Here could be some comment why we need to go unsafe and why it was the best choice.

Reply:
So using `static mut` by default requires `unsafe` blocks, so I left a comment before the declaration statement for this variable.
Done